File: leetcode_27/solution.py
class Solution:
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        # i = 0
        # while i < len(nums):
        #     if nums[i] == val:
        #         del nums[i]
        #     else:
        #         i += 1
        # return len(nums)

        # begin = 0
        # for i in range(len(nums)):
        #     if nums[i] != val:
        #         nums[begin] = nums[i]
        #         begin += 1
        # return begin

File: leetcode_27/solution.go
package main

import "fmt"

func removeElement(nums []int, val int) int {
	i := 0
	for i < len(nums) {
		if nums[i] != val {
			i++
		} else {
			nums = append(nums[:i], nums[i+1:]...)
		}
	}
	return len(nums)
}

func main()  {
	nums := []int{0, 1, 2, 2, 3, 0, 4, 2}
	fmt.Println(removeElement(nums, 2))
}